### PR TITLE
IDE-744 Repository find not limiting results to subfolder when selected

### PR DESCRIPTION
**Pull request by dehilsterlexis**

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

File: comms/DiskRepository.cpp [hunk 1]
```diff
@@ -138,6 +138,10 @@ class CDiskRepository : public CRepositoryBase
                 pos = moduleName.rfind(_T("\\"));
                 moduleName = moduleName.substr(0, pos);
                 algo::replace_all(moduleName, _T("\\"), _T("."));
+                if (!module.empty() && !boost::starts_with(moduleName, module))
```
Comment (GordonSmith) [inline comms/DiskRepository.cpp]: Is "starts_with" appropriate here?

Comment (dehilsterlexis) [reply comms/DiskRepository.cpp]: It is correct. See below:

findstartswith

Comment (dehilsterlexis) [reply comms/DiskRepository.cpp]: @GordonSmith we can restrict it to the selected folder only and no subfolders of the selected folder. Is that the behavior we want or do we want to include the subfolder files as well?

**Comment by dehilsterlexis** [comment]
@GordonSmith what behavior do we want: only files in that folder or all folders that match in all subfolders as well?

**Comment by GordonSmith** [comment]
Subfolders as well (or add a checkbox to make that optional).

**Comment by dehilsterlexis** [comment]
@GordonSmith back to you.  Putting in a subfolder's checkbox should be separate in my opinion and it seems a bit too picky for now. If people want it, we can easily add it in a separate PR.

File: comms/DiskRepository.cpp [hunk 2]
```diff
@@ -138,6 +138,10 @@ class CDiskRepository : public CRepositoryBase
                 pos = moduleName.rfind(_T("\\"));
                 moduleName = moduleName.substr(0, pos);
                 algo::replace_all(moduleName, _T("\\"), _T("."));
+                if (!module.empty() && !boost::equal(moduleName, module) && !boost::starts_with(moduleName, module + _T(".")))
```
Comment (GordonSmith) [inline comms/DiskRepository.cpp]: What is the "starts_with" used for?

Comment (dehilsterlexis) [reply comms/DiskRepository.cpp]: @GordonSmith it is used for making sure we math only Test with "Test." and not "Tester."

Comment (GordonSmith) [reply comms/DiskRepository.cpp]: ok

**Pull request merged**